File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.js
function solution(n) {
    
    if(n<=2) return 1;
    
    var left = 1;
    var right = 2;
    var sum = left + right;
    var answer = 0;
    
    while(left <= right){
        
        if(sum <= n){
            if(sum == n) answer ++;
            right++;
            sum += right;
        }else{
            sum -= left;
            left++
        }
    }
    
    return answer;
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        
        int left = 1;
        int right = 2;
        int sum = left+right;
        int answer = 0;
        
        if(n <= 2) return 1;
        
        while(left <= right){
            if(sum <= n){
                if(sum == n) answer++;
                right++;
                sum += right;
            }else{
                sum -= left;
                left++;
            }
        }
       
        return answer;
    }
}